### PR TITLE
feat: token endpoint

### PR DESCRIPTION
**Pull request by michaelmajgaard**

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/TokenController.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    {
+        _externalTokenValidator = externalTokenValidator;
+        _signingKeyRing = signingKeyRing;
+        _configuration = configuration;
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    [AllowAnonymous]
+    [Route("v2.0/.well-known/openid-configuration")]
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/TokenController.cs:63]: I Think there might be two Routes that should works here, i believe that the "v2.0" is only appended when MSAL is in Azure AD Mode, while that is the case for us, we should think about if we want to support the normal OIDC spec which omits the "v2.0" from the url.

More info can be found at: https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2389 and https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/1824

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs [hunk 2]
```diff
@@ -47,6 +50,8 @@ public sealed class Startup : Common.StartupBase
         public Startup(IConfiguration configuration)
         {
             _configuration = configuration;
+            LogHelper.Logger = new IdentityLogger(); //TODO:remove
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs:53]: Should this be removed before merging, or is it expected to be removed in a later PR?

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs [hunk 3]
```diff
@@ -69,7 +74,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             if (_configuration.GetSetting(Settings.RolesValidationEnabled))
             {
-                app.UseUserMiddleware<FrontendUser>();
+                // app.UseUserMiddleware<FrontendUser>();
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs:77]: Should be deleted if it is no longer needed.

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs [hunk 4]
```diff
@@ -47,6 +50,8 @@ public sealed class Startup : Common.StartupBase
         public Startup(IConfiguration configuration)
         {
             _configuration = configuration;
+            LogHelper.Logger = new IdentityLogger(); //TODO:remove
+            AuthenticationExtensions.DisableHttpsConfiguration = true;
```
Comment (FirestarJes) [inline source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs:54]: This seems like we shouldn't be disabling this, unless this is only for testing and will be removed in a future PR.

Comment (michaelmajgaard) [reply source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs]: fixed

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs (hunk 3 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs]: fixed

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs (hunk 2 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Startup.cs]: fixed

File: source/Energinet.DataHub.MarketParticipant.Application/Commands/GetUserPermissionsResponse.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using Energinet.DataHub.Core.App.Common.Security;
+
+namespace Energinet.DataHub.MarketParticipant.Application.Commands;
+
+public sealed record GetUserPermissionsResponse(IEnumerable<Permission> Permissions);
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Application/Commands/GetUserPermissionsResponse.cs:20]: Test that new unknown permissions do not crash the middleware.

File: source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/TokenController.cs (hunk 1 above)
Comment (michaelmajgaard) [reply source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/TokenController.cs:63]: @FirestarJes warrants a discussion with the team I think :D

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.EntryPoint.WebApi/Controllers/TokenController.cs:63]: @michaelmajgaard  Agree :)

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**